odoo/addons/base/models/ir_ui_menu.py

```diff
@@ -314,3 +315,44 @@ def _get_menuitems_xmlids(self):
             menu.res_id: menu.complete_name
             for menu in menuitems
         }
+
+    @api.model
+    @tools.ormcache()
+    def _get_model_to_menu_root_info(self):
```

Comment:
Can we sort them here with a ordereddict ? (or with a list of tuple)

it would make the code below simpler

also I would give the model in argument to not have a nested structure in the return (and return only the dict we need)

Reply:
The idea is to have _get_model_to_menu_root_info cached for all the users and all model while _get_best_menu_root_for_model uses that info to provide a tailored value for a given user (using the menu he can see). 

Before taking the menu root with the minimal order value, we assemble all menus root info related to the model that the user can see (through the menu he can see). So I think it is not possible to pre-process the order in _get_model_to_menu_root_info. 

What do you think ? Did I miss something ?

Otherwise adding a parameter res_model to _get_model_to_menu_root_info might simplify a bit as we delegate the dict to the cache (but we a little more queries). I can change that way if you think it is worth.

Reply:
> Before taking the menu root with the minimal order value, we assemble all menus root info related to the model that the user can see (through the menu he can see). So I think it is not possible to pre-process the order in _get_model_to_menu_root_info.

are you sure ? isn't this just sorted by `order_idx` ? (so accessible from here)